monash-human-power/dashboard
# Add Storybook

## Description

This PR adds [Storybook](https://storybook.js.org/) to our repo. This was a move done with fairly little consultation (in fact I only found out about its existence on Friday) so feel free to argue against it. That said, it seems really cool/useful and I reckon could help us design the V3 dasboard.

If you want to learn about Storybook, I highly recommend reading their [React tutorial](https://www.learnstorybook.com/intro-to-storybook/react/en/get-started/). You should be safe to stop reading when they start talking about Redux.

In this PR I made stories for Storybook for some of the components in the Logs page, mainly because it's the simplest page that will be relevant for V3. I made a new component `LogFileList`, which used to just be part of `DownloadFilesView`. This was because I felt that for components to play nicely with Storybooks, they have to have all their data passed in as props. @haotongwang has pointed out that this is called a "presentational component" (sorry that's probably common knowledge but I'm a noob). Anyway let me know what you think about this in particular, whether there's something else I should be doing or whether refactoring out presentational components is a fine thing to do.

## Screenshots

The Storybook interface looks something like this:

![image](https://user-images.githubusercontent.com/17876556/94497820-b81a7600-023b-11eb-896b-9f15134a03fe.png)

## Steps to Test

You'll need to `yarn install` in `client` to get the latest packages. Then,
- Check that everything, especially the log file viewer, still works as expected by running the dashboard locally as usual
- Run `yarn run storybook` in `client` to start Storybook. It should open a browser automatically. Have a browse around and see what you think of stuff!
